File: backend/internal/modules/notifications/domain/services/notificationService.go
package services

import (
	"github.com/javierjpv/edenBooks/internal/modules/notifications/application/dto"
	"github.com/javierjpv/edenBooks/internal/modules/notifications/domain/entities"
)

type NotificationService interface {
	CreateNotification(notification dto.NotificationRequest) error

	UpdateNotification(id uint, notification dto.NotificationRequest) error

	DeleteNotification(id uint) error

	GetNotificationByID(id uint) (*entities.Notification, error)

	GetFilteredNotifications(filters map[string]string) ([]entities.Notification, error)

	ListenOrderCreated()
}

File: frontend/edenBooks/src/modules/notifications/services/notificationService.ts
import { INotificationService } from "../interfaces/INotificationService";
import { notificationApi } from "../api/notificationApi";
import { INotificationRequest } from "../interfaces/INotificationRequest";

export const notificationService: INotificationService = {
  CreateNotification: async (notification: INotificationRequest) =>notificationApi.CreateNotification(notification),
  GetNotifications: async () => notificationApi.GetNotifications(),
  GetNotificationById: async (id: number) => notificationApi.GetNotificationById(id),
  UpdateNotification: async (id: number, notification: INotificationRequest) =>notificationApi.UpdateNotification(id, notification),
  DeleteNotification: async (id: number) => notificationApi.DeleteNotification(id),
};
